ntoskrnl/ex/sysinfo.c

```diff
@@ -1391,12 +1387,13 @@ QSI_DEF(SystemHandleInformation)
 
                         HandleInformation->Handles[Index].GrantedAccess =
                             HandleTableEntry->GrantedAccess;
-
-                        ++Index;
                     }
 
                     /* Unlock it */
                     ExUnlockHandleTableEntry(HandleTable, HandleTableEntry);
+
+                    /* Increase count of handles, always */
+                    ++Index;
```

Comment:
`Index` is exclusively used for the index into the `Handles` array of the user-provided `HandleInformation` buffer. Please keep the original intent of the code and separate that from the counting of handles (the `NumberOfHandles` variable) that is also done separately, whatever the size of the user-provided information buffer.

Reply:
I will revert that, though the 2 variables feel redundant after the counting change.